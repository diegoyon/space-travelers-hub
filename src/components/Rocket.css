.rocket-card img {
  width: 15rem;
}

.rocket-card {
  display: flex;
  gap: 1rem;
  margin: 1rem;
}

.details {
  display: flex;
  flex-direction: column;
  gap: 1rem;
}

.details button {
  padding: 1rem;
  background-color: blue;
  color: white;
  width: fit-content;
  border: none;
  border-radius: 5px;
  cursor: pointer;
}
