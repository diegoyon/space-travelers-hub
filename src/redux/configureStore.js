import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missions';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
  },
});

export default store;
