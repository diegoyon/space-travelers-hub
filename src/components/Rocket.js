/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import './Rocket.css';
import { useDispatch } from 'react-redux';
import { toggleReserve } from '../redux/rockets/rocketsSlice';

function Rocket(props) {
  const {
    id, rocket_name, description, flickr_images, reserved,
  } = props;
  const dispatch = useDispatch();

  let content;
  if (reserved) {
    content = <p>reserved</p>;
  }

  return (
    <div className="rocket-card">
      <img src={flickr_images[0]} alt={rocket_name} />
      <div className="details">
        <h2>{rocket_name}</h2>
        {content}
        <p>{description}</p>
        <button type="button" onClick={() => { dispatch(toggleReserve(id)); }}>Reserve Rocket</button>
      </div>
    </div>
  );
}

Rocket.propTypes = {
  id: PropTypes.number.isRequired,
  rocket_name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  flickr_images: PropTypes.arrayOf(PropTypes.string).isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default Rocket;
