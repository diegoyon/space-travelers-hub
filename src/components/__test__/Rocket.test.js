import { render } from '@testing-library/react';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import renderer from 'react-test-renderer';
import Rocket from '../Rocket';

describe(Rocket, () => {
  const initialState = {};
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Rocket
          key={1}
          rocket_name="Rocket Name"
          description="Rocket Description"
          flickr_images={['image1', 'image2']}
          id={1}
          reserved
        />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('rocket displays correct name information', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Rocket
          key={1}
          rocket_name="Rocket Name"
          description="Rocket Description"
          flickr_images={['image1', 'image2']}
          id={1}
          reserved
        />
      </Provider>,
    );
    const rocketName = getByTestId('name').textContent;
    expect(rocketName).toBe('Rocket Name');
  });

  it('rocket displays correct description information', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Rocket
          key={1}
          rocket_name="Rocket Name"
          description="Rocket Description"
          flickr_images={['image1', 'image2']}
          id={1}
          reserved
        />
      </Provider>,
    );
    const rocketDescription = getByTestId('description').textContent;
    expect(rocketDescription).toBe('Rocket Description');
  });
});
