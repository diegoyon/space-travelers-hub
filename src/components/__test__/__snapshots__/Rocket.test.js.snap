// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket renders correctly 1`] = `
<div
  className="rocket-card"
>
  <img
    alt="Rocket Name"
    src="image1"
  />
  <div
    className="details"
  >
    <h2
      data-testid="name"
    >
      Rocket Name
    </h2>
    <div>
      <span
        className="badge"
      >
        Reserved
      </span>
      <span
        data-testid="description"
      >
        Rocket Description
      </span>
    </div>
    <button
      className="cancel"
      onClick={[Function]}
      type="button"
    >
      Cancel Reservation
    </button>
  </div>
</div>
`;
